subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -599,25 +616,7 @@ static void prov_retransmit(struct k_work *work)
 		return;
 	}
 
-	for (i = 0; i < ARRAY_SIZE(link.tx.buf); i++) {
-		struct net_buf *buf = link.tx.buf[i];
-
-		if (!buf) {
-			break;
-		}
-
-		if (BT_MESH_ADV(buf)->busy) {
-			continue;
-		}
-
-		BT_DBG("%u bytes: %s", buf->len, bt_hex(buf->data, buf->len));
-
-		if (i + 1 < ARRAY_SIZE(link.tx.buf) && link.tx.buf[i + 1]) {
-			bt_mesh_adv_send(buf, NULL, NULL);
-		} else {
-			bt_mesh_adv_send(buf, &buf_sent_cb, NULL);
-		}
-	}
```

Comment:
Needs to be reinstated due to not closing link completely. Most notably when using the mesh provisioning functionality.
The first node works correctly then the prov_link flag remains "LINK_ACTIVE"